### PR TITLE
feat(flags): add GET feature_flags/:id/status for determining flag staleness

### PR DESCRIPTION
**Pull request by havenbarnes**
## Problem

This serves as a starting point for https://github.com/PostHog/posthog/issues/16497
By hitting this new endpoint in the Feature Flag > Usage tab, we can show users when their flag isn't being used anymore and is ripe for deletion from Posthog AND in their codebase. Eventually, we can use this status checking logic as a way to automate the archival of flags or give users helpful hints about flag-related tech debt in general

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

- Hit API locally a ton
- Added exhaustive API test cases

### CODE REVIEW

File: posthog/models/feature_flag/flag_status.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+        multivariate = flag.filters.get("multivariate", None)
+        if multivariate:
+            is_multivariate_flag_fully_enabled, fully_enabled_variant_name = self.is_multivariate_flag_fully_enabled(
+                flag
+            )
+        if multivariate and is_multivariate_flag_fully_enabled:
+            return True, f'Currently, this flag will always use the variant "{fully_enabled_variant_name}"'
+        elif self.is_boolean_flag_fully_enabled(flag):
+            return True, 'Currently, this boolean flag will always evaluate to "true"'
```
Comment (dmarticus) [inline posthog/models/feature_flag/flag_status.py:89]: do we need "Currently"?  Feels like that's a superfluous word.

File: posthog/models/feature_flag/flag_status.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        # See if the flag is set to 100% on one variant (or 100% on rolled out and active if boolean flag).
+        is_flag_fully_enabled, fully_enabled_explanation = self.is_flag_fully_enabled(flag)
+        if is_flag_fully_enabled:
+            return FeatureFlagStatus.STALE, fully_enabled_explanation
+
+        # Final, and most expensive check: see if the flag has been evaluated recently.
+        if self.is_flag_unevaluated_recently(flag):
+            return FeatureFlagStatus.INACTIVE, "Flag has not been evaluated recently"
+
+        return FeatureFlagStatus.ACTIVE, "Flag is not fully rolled out and may still be actively called"
```
Comment (dmarticus) [inline posthog/models/feature_flag/flag_status.py:56]:
```suggestion
        return FeatureFlagStatus.ACTIVE, "Flag is not fully rolled out and may still be active"
```

**Review by dmarticus — COMMENTED** [review]
few copy suggestions and one question about the query.

File: posthog/models/feature_flag/flag_status.py (hunk 1 above)
Comment (havenbarnes) [reply posthog/models/feature_flag/flag_status.py:89]: Yep fair

File: posthog/models/feature_flag/flag_status.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            properties=[
+                {
+                    "key": "$feature_flag",
+                    "operator": "exact",
+                    "type": "event",
+                    "value": flag.key,
+                }
+            ],
+            select=[
+                "if(toString(properties.$feature_flag_response) IN ['1', 'true'], 'true', 'false') -- Feature Flag Response"
```
[the remark this answers is not in the document]
Comment (havenbarnes) [reply posthog/models/feature_flag/flag_status.py]: Removed

File: posthog/models/feature_flag/flag_status.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        if is_flag_fully_enabled:
+            return FeatureFlagStatus.STALE, fully_enabled_explanation
+
+        # Final, and most expensive check: see if the flag has been evaluated recently.
+        if self.is_flag_unevaluated_recently(flag):
+            return FeatureFlagStatus.INACTIVE, "Flag has not been evaluated recently"
+
+        return FeatureFlagStatus.ACTIVE, "Flag is not fully rolled out and may still be active"
+
+    def is_flag_fully_enabled(self, flag: FeatureFlag) -> tuple[bool, FeatureFlagStatusReason]:
```
Comment (dmarticus) [inline posthog/models/feature_flag/flag_status.py:58]: I am worried about abstraction collisions between a flag being fully "enabled" (i.e. this flag is rolled out to everyone) vs "enabled" (i.e. "this flag is turned on", or "not disabled").  Feels like we have too many dimensions of status in here that I'd like to disambiguate a bit more somehow.

**Review by dmarticus — COMMENTED** [review]
API design looks fine and idiomatic; love the tests; only have some feedback on what feels like an abstraction collision between a flag being "enabled" | "disabled" vs enabled with a status.  I want to do a better job modeling the different between flags that are enabled vs disabled vs stale vs inactive (especially since I think it'd be cool for folks to just disable flags that are stale and have that "just work", if possible).

File: posthog/models/feature_flag/flag_status.py (hunk 4 above)
Comment (havenbarnes) [reply posthog/models/feature_flag/flag_status.py]: Ah yup good point. "Rolled out" feels much more precise then. I've swapped out all references to "enabled" for "rolled out" in this file and the tests

**Review by dmarticus — APPROVED** [review]

**Pull request merged**